game-core/src/main/java/games/strategy/triplea/delegate/GameStepPropertiesHelper.java

```diff
@@ -137,15 +137,17 @@ public static boolean isNonCombatMove(final GameData data, final boolean doNotTh
   }
 
   /**
-   * Fire rockets after phase is over. Normally would occur after combat move for WW2v2 and WW2v3, and after noncombat
-   * move for WW2v1.
+   * Fire rockets after phase is over. This method is here for legacy support.
+   * Ideally, all maps with rockets will set PROPERY_fireRockets for move and battle phases.
    */
   static boolean isFireRockets(final GameData data) {
     data.acquireReadLock();
     try {
       final String prop = data.getSequence().getStep().getProperties().getProperty(GameStep.PropertyKeys.FIRE_ROCKETS);
       if (prop != null) {
         return Boolean.parseBoolean(prop);
+      } else if (data.getSequence().getStep().getDelegate().getName().equals("battle")) {
+        return Properties.getWW2V2(data) || Properties.getWW2V3(data);
```

Comment:
Not really following this change here?

Reply:
Took me a few minutes to remember. I did most of this stuff a year ago only to find it broke compatibility.

The change is because this PR moves the rocket firing from the end of the MoveDelegate to the BattleDelegate, in part to fix the Scramble bug (where a rocket can target an airbase and prevent a scramble). The exception here is for V1 and compatible maps.